src/db/json.hh

```diff
@@ -182,10 +182,10 @@ namespace json {
             static void wrongType(Type actual, Type requested);
         
             // Throws a runtime_error if the type of the Value does not match the given Type
-            inline void checkType(Type type) const { if (this->type != type) { wrongType(this->type,type); } } 
+            inline void checkType(Type _type) const { if (this->type != _type) { wrongType(this->type, _type); } }
```

Comment:
shadow is my least favorite warning. I was under impression that the scoping was unambiguous and the explicit ```this->``` was also unambiguous at allowing access to the "shadowed" object variables, so it's unclear why this wasn't a warning before and is recently (compilers!?!?!), but this popped up just before fastjson was merged into snoplus rat and similar changes were made there.